### PR TITLE
Test PageActionInvoker logs and fix declaring type name for handler in logs

### PR DESCRIPTION
**Pull request by alefranz**
- Extends test added in #11526 for all the log messages generated by the `PageActionInvoker` on Invoke, similar to what done in #12107 for `ControllerActionInvoker`
 - Updated `ControllerActionInvoker` logs tests to return more meaningful errors on failure.
 - Update the the declaring type name of the executing handler in the logs to use a clean name, similar to what happens for pages and controllers 

/cc @rynowak

### CODE REVIEW

File: src/Mvc/Mvc.Core/test/Infrastructure/ControllerActionInvokerTest.cs [hunk 1]
```diff
@@ -1484,23 +1484,24 @@ public async Task InvokeAsync_Logs()
 
             // Assert
             var messages = testSink.Writes.Select(write => write.State.ToString()).ToList();
+            var actionSignature = $"{typeof(IActionResult).FullName} {nameof(TestController.ActionMethod)}()";
             var controllerName = $"{typeof(ControllerActionInvokerTest).FullName}+{nameof(TestController)} ({typeof(ControllerActionInvokerTest).Assembly.GetName().Name})";
             var actionName = $"{typeof(ControllerActionInvokerTest).FullName}+{nameof(TestController)}.{nameof(TestController.ActionMethod)} ({typeof(ControllerActionInvokerTest).Assembly.GetName().Name})";
             var actionResultName = $"{typeof(CommonResourceInvokerTest).FullName}+{nameof(TestResult)}";
             Assert.Equal(13, messages.Count);
```
Comment (pranavkm) [inline src/Mvc/Mvc.Core/test/Infrastructure/ControllerActionInvokerTest.cs]: Could you rewrite these as:

```C#
Assert.Collection(
    messages,
    m => Assert.Equal($"Route matched with {{}}. Executing controller action with signature {actionSignature} on controller {controllerName}.", m),
    m => Assert.Equal("Execution plan of authorization filters (in the following order): None", m),
    m => Assert.Equal("Execution plan of resource filters (in the following order): None", m),
    m => Assert.Equal("Execution plan of action filters (in the following order): None", m),
    m => Assert.Equal("Execution plan of exception filters (in the following order): None", m),
    m => Assert.Equal("Execution plan of result filters (in the following order): None", m),
    m => Assert.Equal($"Executing controller factory for controller {controllerName}", m),
    m => Assert.Equal($"Executed controller factory for controller {controllerName}", m),
    m => Assert.Equal($"Executing action method {actionName} - Validation state: Valid", m),
    m => Assert.StartsWith($"Executed action method {actionName}, returned result {actionResultName} in ", m),
    m => Assert.Equal($"Before executing action result {actionResultName}.", m),
    m => Assert.Equal($"After executing action result {actionResultName}.", m),
    m => Assert.StartsWith($"Executed action {actionName} in ", m));

File: src/Mvc/Mvc.RazorPages/test/Infrastructure/PageActionInvokerTest.cs [hunk 2]
```diff
@@ -1408,29 +1414,29 @@ public async Task InvokeAction_LogsPageFactory()
...
-            var invoker = CreateInvoker(null, actionDescriptor, logger: logger);
+            var pagePath = actionDescriptor.ViewEnginePath;
+            var methodFullName = hasPageModel
+                ? $"{typeof(PageActionInvokerTest).FullName}+{nameof(TestPageModel)}.{nameof(TestPageModel.OnGetHandler1)}"
+                : $"{typeof(PageActionInvokerTest).FullName}+{nameof(TestPage)}.{nameof(TestPage.OnGetHandler1)}";
+            var methodName = nameof(TestPage.OnGetHandler1);
+            var resultName = typeof(PageResult).FullName;
+            var factoryType = hasPageModel ? "page model" : "page";
+
+            Assert.Equal(13, messages.Count);
```
Comment (pranavkm) [inline src/Mvc/Mvc.RazorPages/test/Infrastructure/PageActionInvokerTest.cs]:
```C#
Assert.Collection(
    messages,
    m => Assert.Equal($"Route matched with {{page = \"foo\"}}. Executing page {pagePath}", m),
    m => Assert.Equal("Execution plan of authorization filters (in the following order): None", m),
    m => Assert.Equal("Execution plan of resource filters (in the following order): None", m),
    m => Assert.Equal("Execution plan of action filters (in the following order): None", m),
    m => Assert.Equal("Execution plan of exception filters (in the following order): None", m),
    m => Assert.Equal("Execution plan of result filters (in the following order): None", m),
    m => Assert.Equal($"Executing {factoryType} factory for page {pageName}", m),
    m => Assert.Equal($"Executed {factoryType} factory for page {pageName}", m),
    m => Assert.Equal($"Executing handler method {methodFullName} - ModelState is Valid", m),
    m => Assert.Equal($"Executed handler method {methodName}, returned result {resultName}.", m),
    m => Assert.Equal($"Before executing action result {resultName}.", m),
    m => Assert.Equal($"After executing action result {resultName}.", m),
    m => Assert.StartsWith($"Executed page {pagePath} in ", m));
```

**Review by pranavkm — COMMENTED** [review]
This looks great. Thanks for the PR!

**Comment by alefranz** [comment]
Updated

File: src/Mvc/Mvc.Core/test/Infrastructure/ControllerActionInvokerTest.cs (hunk 1 above)
Comment (alefranz) [reply src/Mvc/Mvc.Core/test/Infrastructure/ControllerActionInvokerTest.cs]: Sure

File: src/Mvc/Mvc.RazorPages/test/Infrastructure/PageActionInvokerTest.cs (hunk 2 above)
Comment (alefranz) [reply src/Mvc/Mvc.RazorPages/test/Infrastructure/PageActionInvokerTest.cs]: Sure

**Review by pranavkm — APPROVED** [review]

**Comment by pranavkm** [comment]
Thanks for the PR!

**Comment by alefranz** [comment]
Thanks Pranav for reviewing and merging!

**Pull request merged**